File: pinyougou_user_web/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller,uploadService,userService){
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		userService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		userService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		userService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	}
	
	//保存 
	$scope.save=function(){

		//判断密码
		if($scope.entity.password != $scope.password){
			alert("密码不一致，请重新输入...");
			return ;
		}

		var serviceObject;//服务层对象  				
		if($scope.entity.id!=null){//如果有ID
			serviceObject=userService.update( $scope.entity ); //修改  
		}else{
			serviceObject=userService.add( $scope.entity ,$scope.smsCode );//增加
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					//注册成功后，跳转到登录页面(修改：跳转到个人中心的首页)
					location.href="/home-index.html";
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.dele=function(){			
		//获取选中的复选框			
		userService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){			
		userService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}

	//发送短信的方法
	$scope.sendSms=function(){
        userService.sendSms($scope.entity.phone).success(function (response) {
            if(response.success){
            	alert("验证码已发送");
			} else{
            	alert(response.message);
			}
        })
	}

	//用户首页初始化，获取当前登录者用户信息
	$scope.showName=function(){
		userService.showName().success(function (response) {
			$scope.userMap = response;
        })
	}


    $scope.findSeckill=function () {
        userService.findSeckill().success(function (response) {
            $scope.list=response;
        })
    }


    //定义一个数组：把数组根据索引指定状态
    $scope.statusList=["未付款","已付款","未发货","已发货","交易成功","交易关闭","待评价"];
    $scope.statusList2=["立即付款","等待发货","催商家发货","确认收货","评价"];
    $scope.findDingdan=function () {
        userService.findDingdan().success(function (response) {
            $scope.list=response;

        })
    }







    $scope.myInFo=function(){
        userService.myInFo().success(function (response) {
            $scope.userInFo = response;
            $scope.entityUser= $scope.userInFo;
        })
    }
    $scope.address=function(){
        userService.address().success(function (response) {
            $scope.addressList = response;
        })
    }
    $scope.edit1=function(addressedit1){

            $scope.addressedit1 = addressedit1;

    }


	$scope.updeteUser=function(){
        userService.updeteUser($scope.entityUser).success(function (response) {
            if(response.success){
                //注册成功后，跳转到登录页面(修改：跳转到个人中心的首页)
                location.href="/home-index.html";
            }else{
                alert(response.message);
            }
        })
    }


    $scope.uploadFile=function(){
        uploadService.uploadFile().success(function (response) {
            if(response.success){
                $scope.entityUser.user.headPic=response.message;
            }
        })
    }

});	

File: pinyougou_user_web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.common.PageResult;
import com.pinyougou.common.PhoneFormatCheckUtils;
import com.pinyougou.common.Result;
import com.pinyougou.pojo.TbAddress;
import com.pinyougou.pojo.TbUser;
import com.pinyougou.pojo.vo.UserInFo;
import com.pinyougou.user.service.UserService;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * controller
 *
 * @author Administrator
 */
@RestController
@RequestMapping("/user")
public class UserController {

    @Reference(timeout = 30000)
    private UserService userService;

    /**
     * 返回全部列表
     *
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbUser> findAll() {
        return userService.findAll();
    }


    /**
     * 返回全部列表
     *
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page, int rows) {
        return userService.findPage(page, rows);
    }

    /**
     * 增加
     *
     * @param user
     * @return
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbUser user, String smsCode) {

        //验证手机号
        if (!PhoneFormatCheckUtils.isPhoneLegal(user.getPhone())) {
            return new Result(false, "请输入正确的手机号");
        }
        //验证验证码
        if (!userService.checkSmsCode(user.getPhone(), smsCode)) {
            return new Result(false, "验证码不正确");
        }
        try {
            userService.add(user);
            return new Result(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "增加失败");
        }
    }

    /**
     * 修改
     *
     * @param user
     * @return
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbUser user) {
        try {
            userService.update(user);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    /**
     * 获取实体
     *
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbUser findOne(Long id) {
        return userService.findOne(id);
    }

    /**
     * 批量删除
     *
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            userService.delete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }
    }

    /**
     * 查询+分页
     *
     * @param
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbUser user, int page, int rows) {
        return userService.findPage(user, page, rows);
    }

    @RequestMapping("sendSms")
    public Result sendSms(String phone) {
        try {
            userService.sendSms(phone);
            return new Result(true, "操作成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "操作失败");
        }
    }

    //获取当前登录者用户信息的
    @RequestMapping("showName")
    public Map showName() {
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        Map map = new HashMap();
        map.put("userName", name);
        return map;
    }

    @RequestMapping("findDingdan")
    public List<Map> findDingdan(){
        String name =SecurityContextHolder.getContext().getAuthentication().getName();
        if (!"anonymousUser".equals(name)) {
            new RuntimeException("请先登录");
        }
        List<Map> dingdan = userService.findDingdan(name);
        return dingdan;
    }
    @RequestMapping("findSeckill")
    public List<Map> findSeckill(){
        String name =SecurityContextHolder.getContext().getAuthentication().getName();
        if (!"anonymousUser".equals(name)) {
            new RuntimeException("请先登录");
        }
        List<Map> dingdan = userService.findSeckill(name);
        return dingdan;
    }





    @RequestMapping("/myInFo")
    public UserInFo myInFo() {
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        UserInFo userInFo = userService.myInFo(name);
        return userInFo;
    }

    @RequestMapping("/updeteUser")
    public Result updateUser(@RequestBody UserInFo userInFo) {
        try {
            userService.updateUser(userInFo);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    @RequestMapping("/address")
    public List<TbAddress> address() {
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        List<TbAddress> addressList = userService.address(name);
        return addressList;
    }


}
